### PR TITLE
Support python 3.13

### PR DESCRIPTION
**Pull request by Firdous2307**
# Add support for Python 3.13

This PR adds support for Python 3.13 to the icalendar project. It includes updates to the test configurations, project metadata, and documentation.

## Changes made:
1. Added Python 3.13 to the GitHub Actions test matrix
2. Included py313 environment in tox configuration
3. Updated classifiers in pyproject.toml to include Python 3.13
4. Updated README with Python 3.13 compatibility information
5. Added documentation on the process of updating Python versions

## Documentation:
- [x] Updated README with new Python version support
- [x] Added section in `docs/maintenance.rst` about updating Python versions

This PR addresses issue #723.

----
📚 Documentation preview 📚: https://icalendar--729.org.readthedocs.build/

### CODE REVIEW

**Comment by niccokunzmann** [comment]
@Firdous2307 Thanks! I wonder why there are no merge conflicts.. I will update the branch.

Could you fix the changelog? The last change was in the already released version.
Thanks for the documentation!

**Comment by Firdous2307** [comment]
@niccokunzmann no problem , I have fixed the changelog

File: CHANGES.rst [hunk 1]
```diff
@@ -1,30 +1,19 @@
...
 
 Breaking changes:
 
 - ...
 
 New features:
 
-- ...
-
-Bug fixes:
```
Comment (niccokunzmann) [inline CHANGES.rst]: These lines need to  stay... That is where people fill things in until we release.

File: CHANGES.rst [hunk 2]
```diff
@@ -1,30 +1,19 @@
...
-
-- ...
-
-6.0.1 (2024-10-13)
-------------------
-
-New features:
-
 - Added ``Event.end``, ``Event.start``, ``Event.dtstart``, and ``Event.dtend`` attributes. See `Issue 662 <https://github.com/collective/icalendar/issues/662>`_.
 - Test compatibility with Python 3.13
```
Comment (niccokunzmann) [inline CHANGES.rst]: that would need removing

File: docs/maintenance.rst [hunk 3]
```diff
@@ -148,3 +148,16 @@ This section contains useful links for maintainers and collaborators:
 
 - `Future of icalendar, looking for maintainer #360 <https://github.com/collective/icalendar/discussions/360>`__
 - `Comment on the Plone tests running with icalendar <https://github.com/collective/icalendar/pull/447#issuecomment-1277643634>`__
+
```
Comment (niccokunzmann) [inline docs/maintenance.rst:151]: yes, I like it. This is just about the changelog now.

File: CHANGES.rst [hunk 4]
```diff
@@ -1,30 +1,19 @@
 Changelog
 =========
 
-6.0.2 (unreleased)
```
Comment (niccokunzmann) [inline CHANGES.rst]: no, that needs to stay... We are creating changes for a new release now.

File: CHANGES.rst [hunk 5]
```diff
@@ -1,30 +1,19 @@
...
+6.0.1 (unreleased)
 ------------------
 
 Minor changes:
 
-- ...
+- Test compatibility with Python 3.13
```
Comment (niccokunzmann) [inline CHANGES.rst:9]: nice, thanks!

**Review by niccokunzmann — CHANGES_REQUESTED** [review]
Thanks! I added a few comments to clear possible confusion about what I meant. 6.0.1 is already released. We cannot change the changelog for that except take that one line about 3.13 out that was accidentally not in 6.0.2 but added to 6.0.1.

**Comment by niccokunzmann** [comment]
Thanks for working on it! It is nice for me to collaborate instead of doing it myself! I hope I learn to be better at communicating this way, too :)

File: CHANGES.rst (hunk 4 above)
Comment (mohammedfirdouss) [reply CHANGES.rst]: Okay noted

File: docs/maintenance.rst (hunk 3 above)
Comment (mohammedfirdouss) [reply docs/maintenance.rst:151]: Nice

**Comment by niccokunzmann** [comment]
Please let me know when you would like me to review/merge!

**Comment by Firdous2307** [comment]
> Thanks for working on it! It is nice for me to collaborate instead of doing it myself! I hope I learn to be better at communicating this way, too :)

Thank you! Glad you're enjoying the collaboration. You're doing great, and it only gets better with practice! 😊

**Comment by Firdous2307** [comment]
@niccokunzmann It's ready for review!!

**Comment by Firdous2307** [comment]
@niccokunzmann It's ready for review!!

File: CHANGES.rst [hunk 6]
```diff
@@ -2,11 +2,15 @@ Changelog
 =========
 
 6.0.2 (unreleased)
+
+
+6.0.1 (unreleased)
```
Comment (niccokunzmann) [inline CHANGES.rst]:
```suggestion
6.0.1 (2024-10-13)
```

Comment (niccokunzmann) [reply CHANGES.rst]: Have a look, you can commit suggestions...

**Comment by niccokunzmann** [comment]
Please have a look at the changelog file
https://github.com/collective/icalendar/blob/30197304a45053dc4cb31e279b992cb45c46d5e1/CHANGES.rst
You can view this file and you will see inconsistencies at the start.

your changes should be added to the top heading:

https://github.com/collective/icalendar/blob/main/CHANGES.rst

Everything else would be left untouched.

**Comment by niccokunzmann** [comment]
Hm... No problem if you want to not spend so much time just on the changelog... I create a PR for you ....

**Comment by niccokunzmann** [comment]
I could not create a PR but I could commit directly.

I think, https://github.com/collective/icalendar/pull/730#discussion_r1798599889 is valid for this PR, though, could you use the proper RST way and then, I think, it is fine to merge!

**Comment by stevepiercy** [comment]
The changes in this PR do not do what the description says.

**Comment by niccokunzmann** [comment]
@stevepiercy I think that is ok. @Firdous2307 is contributing for the first time and it does not need to be too perfect. This is a follow-up PR from the on made already to close #723. In that sense, it is not too bad. We had the changes in here but they were removed because of duplication.

**Comment by Firdous2307** [comment]
@niccokunzmann Any more changes needed to make?

File: CHANGES.rst [hunk 7]
```diff
@@ -23,10 +23,9 @@ Bug fixes:
 6.0.1 (2024-10-13)
 ------------------
 
-New features:
+Bug fixes:
```
Comment (niccokunzmann) [inline CHANGES.rst]:
```suggestion
New features:
```

File: docs/maintenance.rst [hunk 8]
```diff
@@ -148,3 +148,16 @@ This section contains useful links for maintainers and collaborators:
...
+Updating Python Versions
+------------------------
+
+When adding support for a new Python version or removing support for an old one, the following files need to be updated:
+
+1. `.github/workflows/tests.yml`: Add or remove the Python version from the test matrix.
```
Comment (niccokunzmann) [inline docs/maintenance.rst]:
```suggestion
1. ``.github/workflows/tests.yml``: Add or remove the Python version from the test matrix.
```

File: docs/maintenance.rst [hunk 9]
```diff
@@ -148,3 +148,16 @@ This section contains useful links for maintainers and collaborators:
...
+2. `tox.ini`: Update the `envlist` to include or remove the Python version.
```
Comment (niccokunzmann) [inline docs/maintenance.rst]:
```suggestion
2. ``tox.ini``: Update the ``envlist`` to include or remove the Python version.
```

File: docs/maintenance.rst [hunk 10]
```diff
@@ -148,3 +148,16 @@ This section contains useful links for maintainers and collaborators:
...
+3. `pyproject.toml`: Update the `requires-python` line and the `classifiers` list.
```
Comment (niccokunzmann) [inline docs/maintenance.rst]:
```suggestion
3. ``pyproject.toml``: Update the ``requires-python`` line and the ``classifiers`` list.
```

File: docs/maintenance.rst [hunk 11]
```diff
@@ -148,3 +148,16 @@ This section contains useful links for maintainers and collaborators:
...
+4. `README.rst`: Update the compatibility information.
```
Comment (niccokunzmann) [inline docs/maintenance.rst]:
```suggestion
4. ``README.rst``: Update the compatibility information.
```

File: docs/maintenance.rst [hunk 12]
```diff
@@ -148,3 +148,16 @@ This section contains useful links for maintainers and collaborators:
...
+5. `docs/maintenance.rst`: Update this list if any new files need to be modified.
```
Comment (niccokunzmann) [inline docs/maintenance.rst]:
```suggestion
5. ``docs/maintenance.rst``: Update this list if any new files need to be modified.
```

**Review by niccokunzmann — APPROVED** [review]
I will apply these changes and merge it. Thanks!

**Comment by Firdous2307** [comment]
> I will apply these changes and merge it. Thanks!

@niccokunzmann You are welcome, had a fun experience working on this issue

**Comment by niccokunzmann** [comment]
@Firdous2307 Thanks for your work!

I squashed the commits into one and so you will need to start from collective/icalendar's main branch for new PRs.

**Comment by niccokunzmann** [comment]
If you are doing the hacvktoberfest, have a look if this shows up as a contribution - it should. If it  does not , let me know :)

**Comment by Firdous2307** [comment]
> If you are doing the hacvktoberfest, have a look if this shows up as a contribution - it should. If it does not , let me know :)

@niccokunzmann Yes I am, ideally it shows up as a contribution but they review it in 7 days. I will keep you posted on it.

**Pull request merged**